src/Extensions/Compatibility/OmexCompatibilityInitializationException.cs

```diff
@@ -0,0 +1,14 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+
+namespace Microsoft.Omex.Extensions.Compatibility
+{
+	internal class OmexCompatibilityInitializationException : Exception
+	{
+		internal OmexCompatibilityInitializationException()
+			: base("Compatibility classes were not initialized. Please call AddOmexCompatibilityServices() on HostBuilder")
```

Comment:
Nice. It's great to be clear when you enter a failure case. I've spent too much time trying to diagnose generic failures 🙂

Reply:
Thanks, I also like clarity